wsagent/agent/src/main/resources/installers/1.0.2/org.eclipse.che.ws-agent.script.sh

```diff
@@ -0,0 +1,286 @@
+        ${SUDO} yum -y install ${PACKAGES};
+    }
+
+    if [ ${INSTALL_JDK} = true ]; then
+        ln -s /usr/lib/jvm/java-1.8.0-openjdk $JAVA_HOME
+    fi
+
+# openSUSE 13.2
+###############
+elif echo ${LINUX_TYPE} | grep -qi "opensuse"; then
+
+    if [ ${INSTALL_JDK} = true ]; then
+        PACKAGES=${PACKAGES}" java-1_8_0-openjdk-devel";
+    fi
+
+    test "${PACKAGES}" = "" || {
+        ${SUDO} zypper install -y ${PACKAGES};
+    }
+
+    if [ ${INSTALL_JDK} = true ]; then
+        ln -s /usr/lib/jvm/java-1.8.0-openjdk $JAVA_HOME
+    fi
+
+
+# Alpine 3.3
+############$$
+elif echo ${LINUX_TYPE} | grep -qi "alpine"; then
+
+    if [ ${INSTALL_JDK} = true ]; then
+        PACKAGES=${PACKAGES}" openjdk8";
+    fi
+
+    test "${PACKAGES}" = "" || {
+        ${SUDO} apk update
+        ${SUDO} apk add ${PACKAGES};
+    }
+
+    # Link OpenJDK to JAVA_HOME
+    if [ ${INSTALL_JDK} = true ]; then
+        ln -s /usr/lib/jvm/java-1.8-openjdk $JAVA_HOME
+    fi
+
+# Centos 6.6, 6.7, 6.8
+############
+elif echo ${LINUX_TYPE} | grep -qi "CentOS"; then
+
+     if [ ${INSTALL_JDK} = true ]; then
+        PACKAGES=${PACKAGES}" java-1.8.0-openjdk-devel";
+     fi
+
+     test "${PACKAGES}" = "" || {
+         ${SUDO} yum -y install ${PACKAGES};
+     }
+
+     if [ ${INSTALL_JDK} = true ]; then
+         ln -s /usr/lib/jvm/java-1.8.0-openjdk $JAVA_HOME
+     fi
+
+# Red Hat Enterprise Linux 6
+############################
+elif echo ${LINUX_TYPE} | grep -qi "Red Hat"; then
+    if [ ${INSTALL_JDK} = true ]; then
+        PACKAGES=${PACKAGES}" java-1.8.0-openjdk-devel.x86_64";
+    fi
+
+    test "${PACKAGES}" = "" || {
+        ${SUDO} yum install -y ${PACKAGES};
+    }
+
+    if [ ${INSTALL_JDK} = true ]; then
+        ln -s /usr/lib/jvm/java-1.8.0-openjdk $JAVA_HOME
+    fi
+
+else
+    >&2 echo "Unrecognized Linux Type"
+    >&2 cat $FILE
+    exit 1
+fi
+
+########################
+### Install ws-agent ###
+########################
+
+rm -rf ${CHE_DIR}/ws-agent
+mkdir -p ${CHE_DIR}/ws-agent
+
+
+# Compute URI of workspace master
+WORKSPACE_MASTER_URI=$(echo $CHE_API | cut -d / -f 1-3)
+
+## Evaluate variables now that prefix is defined
+eval "DOWNLOAD_AGENT_BINARIES_URI=${DOWNLOAD_AGENT_BINARIES_URI}"
+
+
+if [ -f "${LOCAL_AGENT_BINARIES_URI}" ] && [ -s "${LOCAL_AGENT_BINARIES_URI}" ]
+then
+    tar zxf "${LOCAL_AGENT_BINARIES_URI}" -C ${CHE_DIR}/ws-agent
+else
+    echo "Workspace Agent will be downloaded from Workspace Master"
+    AGENT_BINARIES_URI=${DOWNLOAD_AGENT_BINARIES_URI}
+    if [ ${CURL_INSTALLED} = true ]; then
+      curl -s  ${AGENT_BINARIES_URI} | tar  xzf - -C ${CHE_DIR}/ws-agent
+    else
+      # replace https by http as wget may not be able to handle ssl
+      AGENT_BINARIES_URI=$(echo ${AGENT_BINARIES_URI} | sed 's/https/http/g')
+
+      # use wget
+      wget -qO- ${AGENT_BINARIES_URI} | tar xzf - -C ${CHE_DIR}/ws-agent
+    fi
+
+fi
+
+if [ -z "${CHE_WORKSPACE_WSAGENT_JAVA_OPTIONS}" ]; then
```

Comment:
@skabashnyuk can you clarify how ws-agent v 1.0.1 would behave after we merge this PR?

Reply:
it will work. Tomcat will take ```JAVA_OPTS```.